### ISSUE TITLE
Error(s) in loading state_dict for PoseResNet: Missing key(s) in state_dict

### ISSUE THREAD

**Issue by lxltaibo** [Microsoft/human-pose-estimation.pytorch#11]
When I run "python pose_estimation/valid.py --cfg experiments/coco/resnet50/256x192_d256x3_adam_lr1e-3.yaml --flip-test --model-file models/pytorch/pose_coco/pose_resnet_50_256x192.pth.tar", it reported an error:
=> loading model from models/pytorch/pose_coco/pose_resnet_50_256x192.pth.tar
Traceback (most recent call last):
  File "pose_estimation/valid.py", line 165, in <module>
    main()
  File "pose_estimation/valid.py", line 123, in main
    model.load_state_dict(torch.load(config.TEST.MODEL_FILE))
  File "/usr/local/lib/python2.7/dist-packages/torch/nn/modules/module.py", line 721, in load_state_dict
    self.__class__.__name__, "\n\t".join(error_msgs)))
RuntimeError: Error(s) in loading state_dict for PoseResNet:
        Missing key(s) in state_dict: "bn1.num_batches_tracked", "layer1.0.bn1.num_batches_tracked", "layer1.0.bn2.num_batches_tracked", "layer1.0.bn3.num_batches_tracked", "layer1.0.downsample.1.num_batches_tracked", "layer1.1.bn1.num_batches_tracked", "layer1.1.bn2.num_batches_tracked", "layer1.1.bn3.num_batches_tracked", "layer1.2.bn1.num_batches_tracked", "layer1.2.bn2.num_batches_tracked", "layer1.2.bn3.num_batches_tracked", "layer2.0.bn1.num_batches_tracked", "layer2.0.bn2.num_batches_tracked", "layer2.0.bn3.num_batches_tracked", "layer2.0.downsample.1.num_batches_tracked", "layer2.1.bn1.num_batches_tracked", "layer2.1.bn2.num_batches_tracked", "layer2.1.bn3.num_batches_tracked", "layer2.2.bn1.num_batches_tracked", "layer2.2.bn2.num_batches_tracked", "layer2.2.bn3.num_batches_tracked", "layer2.3.bn1.num_batches_tracked", "layer2.3.bn2.num_batches_tracked", "layer2.3.bn3.num_batches_tracked", "layer3.0.bn1.num_batches_tracked", "layer3.0.bn2.num_batches_tracked", "layer3.0.bn3.num_batches_tracked", "layer3.0.downsample.1.num_batches_tracked", "layer3.1.bn1.num_batches_tracked", "layer3.1.bn2.num_batches_tracked", "layer3.1.bn3.num_batches_tracked", "layer3.2.bn1.num_batches_tracked", "layer3.2.bn2.num_batches_tracked", "layer3.2.bn3.num_batches_tracked", "layer3.3.bn1.num_batches_tracked", "layer3.3.bn2.num_batches_tracked", "layer3.3.bn3.num_batches_tracked", "layer3.4.bn1.num_batches_tracked", "layer3.4.bn2.num_batches_tracked", "layer3.4.bn3.num_batches_tracked", "layer3.5.bn1.num_batches_tracked", "layer3.5.bn2.num_batches_tracked", "layer3.5.bn3.num_batches_tracked", "layer4.0.bn1.num_batches_tracked", "layer4.0.bn2.num_batches_tracked", "layer4.0.bn3.num_batches_tracked", "layer4.0.downsample.1.num_batches_tracked", "layer4.1.bn1.num_batches_tracked", "layer4.1.bn2.num_batches_tracked", "layer4.1.bn3.num_batches_tracked", "layer4.2.bn1.num_batches_tracked", "layer4.2.bn2.num_batches_tracked", "layer4.2.bn3.num_batches_tracked", "deconv_layers.1.num_batches_tracked", "deconv_layers.4.num_batches_tracked", "deconv_layers.7.num_batches_tracked". 

Environment:
Pytorch version 0.5.0
CUDA 8.0 and cudnn 6

**Comment by leoxiaobin** [comment]
The lib is tested by pytorch v0.4.0 and v0.4.1, not tested by pytorch v0.5.0. Please switch to v0.4.1 or v0.4.0.
And also it is developed using python3 not python2, it seems your environment is python2.

**Comment by shreyasrajesh** [comment]
Hi, I am having an issue with pytorch v0.4.1 and python3 as below. Any suggestion as to what the issue might be, will be really helpful. Thanks.

```
=> loading model from models/pytorch/pose_coco/pose_resnet_50_384x288.pth.tar
/usr/local/lib/python3.5/dist-packages/torch/nn/functional.py:52: UserWarning: size_average and reduce args will be deprecated, please use reduction='elementwise_mean' instead.
  warnings.warn(warning.format(ret))
loading annotations into memory...
Done (t=0.25s)
creating index...
index created!
=> classes: ['__background__', 'person']
=> num_images: 5000
=> load 6352 samples
OpenCV(3.4.1) Error: Assertion failed (src.cols > 0 && src.rows > 0) in warpAffine, file /io/opencv/modules/imgproc/src/imgwarp.cpp, line 2691
OpenCV(3.4.1) Error: Assertion failed (src.cols > 0 && src.rows > 0) in warpAffine, file /io/opencv/modules/imgproc/src/imgwarp.cpp, line 2691
OpenCV(3.4.1) Error: Assertion failed (src.cols > 0 && src.rows > 0) in warpAffine, file /io/opencv/modules/imgproc/src/imgwarp.cpp, line 2691
OpenCV(3.4.1) Error: Assertion failed (src.cols > 0 && src.rows > 0) in warpAffine, file /io/opencv/modules/imgproc/src/imgwarp.cpp, line 2691
OpenCV(3.4.1) Error: Assertion failed (src.cols > 0 && src.rows > 0) in warpAffine, file /io/opencv/modules/imgproc/src/imgwarp.cpp, line 2691
OpenCV(3.4.1) Error: Assertion failed (src.cols > 0 && src.rows > 0) in warpAffine, file /io/opencv/modules/imgproc/src/imgwarp.cpp, line 2691
Traceback (most recent call last):
  File "pose_estimation/valid.py", line 165, in <module>
OpenCV(3.4.1) Error: Assertion failed (src.cols > 0 && src.rows > 0) in warpAffine, file /io/opencv/modules/imgproc/src/imgwarp.cpp, line 2691
    main()
  File "pose_estimation/valid.py", line 161, in main
OpenCV(3.4.1) Error: Assertion failed (src.cols > 0 && src.rows > 0) in warpAffine, file /io/opencv/modules/imgproc/src/imgwarp.cpp, line 2691
    final_output_dir, tb_log_dir)
  File "/human-pose-estimation.pytorch/pose_estimation/../lib/core/function.py", line 108, in validate
OpenCV(3.4.1) Error: Assertion failed (src.cols > 0 && src.rows > 0) in warpAffine, file /io/opencv/modules/imgproc/src/imgwarp.cpp, line 2691
    for i, (input, target, target_weight, meta) in enumerate(val_loader):
  File "/usr/local/lib/python3.5/dist-packages/torch/utils/data/dataloader.py", line 336, in __next__
    return self._process_next_batch(batch)
  File "/usr/local/lib/python3.5/dist-packages/torch/utils/data/dataloader.py", line 357, in _process_next_batch
    raise batch.exc_type(batch.exc_msg)
cv2.error: Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/torch/utils/data/dataloader.py", line 106, in _worker_loop
    samples = collate_fn([dataset[i] for i in batch_indices])
  File "/usr/local/lib/python3.5/dist-packages/torch/utils/data/dataloader.py", line 106, in <listcomp>
    samples = collate_fn([dataset[i] for i in batch_indices])
  File "/human-pose-estimation.pytorch/pose_estimation/../lib/dataset/JointsDataset.py", line 104, in __getitem__
    flags=cv2.INTER_LINEAR)
cv2.error: OpenCV(3.4.1) /io/opencv/modules/imgproc/src/imgwarp.cpp:2691: error: (-215) src.cols > 0 && src.rows > 0 in function warpAffine
```

**Comment by leoxiaobin** [comment]
@shreyasrajesh , please make sure that you put your data at the right path. The directory should be like this:
```${POSE_ROOT}
|-- data
`-- |-- coco
    `-- |-- annotations
        |   |-- person_keypoints_train2017.json
        |   `-- person_keypoints_val2017.json
        |-- person_detection_results
        |   |-- COCO_val2017_detections_AP_H_56_person.json
        `-- images
            |-- train2017
            |   |-- 000000000009.jpg
            |   |-- 000000000025.jpg
            |   |-- 000000000030.jpg
            |   |-- ... 
            `-- val2017
                |-- 000000000139.jpg
                |-- 000000000285.jpg
                |-- 000000000632.jpg
                |-- ... 
```

**Comment by lxltaibo** [comment]
The problem was solved by install pytorch-0.4.0.
